articles/openshift/oshift-ref-security-best-practices.md

```diff
@@ -0,0 +1,112 @@
+
+  -  Clair
+
+You should also apply the Principle of least privilege with containers too. If it can be avoided you should never run containers as the root user; instead always run as a user with just enough privileges to run the required container workload.
+
+## Secrets
+
+Sensitive information, such as passwords, SSH keys, tokens and certificates, are examples of information that could be considered secret. Such secrets should never be stored in any place in readable format (for example, in source code, GitHub repositories, and so on). It is imperative that when secrets are required in OpenShift, they are set up as OpenShift Secrets. This allows the secrets to be encrypted within OpenShift and accessed only by the relevant containers when they are needed.
+
+## Limiting resources
+
+You should always set up containers with resource limits. In the event of a denial of service attack, or even just a burst of very high load on a service, these resource limits will prevent pods from using all available resources in the node they are running on. You can read more about setting resource quotas and limits [here](https://docs.openshift.com/container-platform/3.11/dev_guide/compute_resources.html).
+
+## Privileged ports
+
+TCP/IP ports below port 1024 are considered to be privileged ports and you should not use them to front any service that is deployed into OpenShift.
+
+## Route security
+
+Within OpenShift, you can expose your services using routes. Creating a route to a service enables your users to access the service through a URL on a browser. By default, the URLs that are created are exposed on HTTP port 80 without any SSL certificate. You can secure these routes with SSL encryption using one of three SSL termination methods:
+
+ - Edge - SSL termination is done on the HAProxy router from which point onwards the traffic is un-encrypted
+
+ - Passthrough termination - There is no SSL decryption on the routers; instead connections are passed straight through the routers to the OpenShift applications, which are then responsible for SSL certificate termination and traffic decryption.
+
+ - Re-Encrypt - This process uses two certificates: the first certificate is held on the HAProxy router and the second in the pods. This secures traffic all the way down to the application.
+
+It is possible to further secure routes by applying IP whitelisting on routes. We have documented the process of IP whitelisting [here](oshift-how-restrict-access-to-openshift-routes-by-ip-address.md).
+
+UKCloud can whitelist individual IP addresses or entire subnets to access the OpenShift console and metrics page. It is also possible to do the same for the services that you create and deploy in OpenShift, even if those services are exposed to various Government Community Networks with the use of security groups. These restrictions can be applied at the time the cluster is deployed or can be done later via a Service Request to the UKCloud OpenShift team.
```

Comment:
Make this a bit clearer, we can whitelist the control plane (which includes the following infrastructure components) and we can also whitelist the dataplanes on any network that is attached. Be clear that this blocks access to all dataplane routes from the given network but is an effective security control.